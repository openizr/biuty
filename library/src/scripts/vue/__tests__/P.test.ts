/**
 * Copyright (c) Matthieu Jabbour. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import UIP from 'scripts/vue/P.vue';
import { mount } from '@vue/test-utils';

describe('vue/UIP', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders correctly - basic', () => {
    const wrapper = mount(UIP, {
      propsData: { label: 'Test', modifiers: 'large' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with id', () => {
    const wrapper = mount(UIP, {
      propsData: { label: 'Test', id: 'test' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with itemProp', () => {
    const wrapper = mount(UIP, {
      propsData: { label: 'Test', itemProp: 'description' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
