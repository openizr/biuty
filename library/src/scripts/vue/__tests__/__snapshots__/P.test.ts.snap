// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vue/UIP renders correctly - basic 1`] = `
<p
  class="ui-p ui-p--large"
>
  Test
</p>
`;

exports[`vue/UIP renders correctly - with id 1`] = `
<p
  class="ui-p"
  id="test"
>
  Test
</p>
`;

exports[`vue/UIP renders correctly - with itemProp 1`] = `
<p
  class="ui-p"
  itemprop="description"
>
  Test
</p>
`;
